common/src/main/cpp/classes/openxr_fb_scene_manager.cpp

```diff
@@ -0,0 +1,390 @@
+
+	ClassDB::bind_method(D_METHOD("set_auto_create", "enable"), &OpenXRFbSceneManager::set_auto_create);
+	ClassDB::bind_method(D_METHOD("get_auto_create"), &OpenXRFbSceneManager::get_auto_create);
+
+	ClassDB::bind_method(D_METHOD("set_visible", "visible"), &OpenXRFbSceneManager::set_visible);
+	ClassDB::bind_method(D_METHOD("get_visible"), &OpenXRFbSceneManager::get_visible);
+	ClassDB::bind_method(D_METHOD("show"), &OpenXRFbSceneManager::show);
+	ClassDB::bind_method(D_METHOD("hide"), &OpenXRFbSceneManager::hide);
+
+	ClassDB::bind_method(D_METHOD("create_scene_anchors"), &OpenXRFbSceneManager::create_scene_anchors);
+	ClassDB::bind_method(D_METHOD("remove_scene_anchors"), &OpenXRFbSceneManager::remove_scene_anchors);
+	ClassDB::bind_method(D_METHOD("are_scene_anchors_created"), &OpenXRFbSceneManager::are_scene_anchors_created);
+
+	ClassDB::bind_method(D_METHOD("request_scene_capture", "request"), &OpenXRFbSceneManager::request_scene_capture, DEFVAL(""));
+	ClassDB::bind_method(D_METHOD("is_scene_capture_enabled"), &OpenXRFbSceneManager::is_scene_capture_enabled);
+
+	ClassDB::bind_method(D_METHOD("get_anchor_uuids"), &OpenXRFbSceneManager::get_anchor_uuids);
+	ClassDB::bind_method(D_METHOD("get_anchor_node", "uuid"), &OpenXRFbSceneManager::get_anchor_node);
+	ClassDB::bind_method(D_METHOD("get_spatial_entity", "uuid"), &OpenXRFbSceneManager::get_spatial_entity);
+
+	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "default_scene", PROPERTY_HINT_RESOURCE_TYPE, "PackedScene"), "set_default_scene", "get_default_scene");
+	ADD_PROPERTY(PropertyInfo(Variant::STRING_NAME, "scene_setup_method", PROPERTY_HINT_NONE, ""), "set_scene_setup_method", "get_scene_setup_method");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "auto_create", PROPERTY_HINT_NONE, ""), "set_auto_create", "get_auto_create");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "visible", PROPERTY_HINT_NONE, ""), "set_visible", "get_visible");
+
+	ADD_SIGNAL(MethodInfo("scene_anchor_created", PropertyInfo(Variant::Type::OBJECT, "scene_node"), PropertyInfo(Variant::Type::OBJECT, "spatial_entity")));
+	ADD_SIGNAL(MethodInfo("scene_data_missing"));
+	ADD_SIGNAL(MethodInfo("scene_capture_completed", PropertyInfo(Variant::Type::BOOL, "success")));
+}
+
+bool OpenXRFbSceneManager::_set(const StringName &p_name, const Variant &p_value) {
+	PackedStringArray parts = p_name.split("/", true, 2);
+	if (parts[0] == "scenes") {
+		const PackedStringArray &semantic_labels = OpenXRFbSceneExtensionWrapper::get_supported_semantic_labels();
+		if (semantic_labels.has(parts[1])) {
+			scenes[parts[1]] = p_value;
+			return true;
+		}
+	}
+	return false;
+}
+
+bool OpenXRFbSceneManager::_get(const StringName &p_name, Variant &r_ret) const {
+	PackedStringArray parts = p_name.split("/", true, 2);
+	if (parts[0] == "scenes") {
+		const PackedStringArray &semantic_labels = OpenXRFbSceneExtensionWrapper::get_supported_semantic_labels();
+		if (semantic_labels.has(parts[1])) {
```

Comment:
I'm curious why we need to check that`semantic_labels` contains the given key. Can't we just check that `scenes[parts[1]] != nullptr`?